packages/node/src/channels.ts

```diff
@@ -0,0 +1,387 @@
+ *    }
+ *  },
+ *  proposedAppInstances: same schema as appInstances,
+ *  freeBalances: Map<AssetType,
+ *    freeBalance: {
+ *      alice: Address,
+ *      aliceBalance: BigNumber,
+ *      bob: Address,
+ *      bobBalance: BigNumber,
+ *      uniqueId: number,
+ *      localNonce: number,
+ *      timeout: number,
+ *      dependencyNonce: {
+ *        isSet: boolean,
+ *        salt: string,
+ *        nonceValue: number
+ *      }
+ *    }
+ *  }
+ * }
+ */
+
+/**
+ * This class is only a type implementation of a channel schema for the
+ * purposes of updating and retrieving a channel's state from the store.
+ *
+ * An instance is by itself stateless and effectively reflects the state of an
+ * according channel in the store.
+ */
+class Channel {
+  constructor(
+    readonly multisigAddress: Address,
+    readonly multisigOwners: Address[],
+    readonly appsNonce: Nonce = new Nonce(true, 0, 0),
+    readonly freeBalances: {
+      [assetType: number]: FreeBalance;
+    } = Channel.initialFreeBalances(multisigOwners, appsNonce),
+    readonly appInstances: {
+      [appInstanceId: string]: AppInstanceInfo;
+    } = {},
+    readonly proposedAppInstances: {
+      [appInstanceId: string]: AppInstanceInfo;
+    } = {}
+  ) {}
+
+  static initialFreeBalances(
+    multisigOwners: Address[],
+    initialAppsNonce: Nonce
+  ): {
+    [assetType: number]: FreeBalance;
+  } {
+    // TODO: extend to all asset types
+    const ethFreeBalance = new FreeBalance(
+      multisigOwners[0],
+      zeroBalance,
+      multisigOwners[1],
+      zeroBalance,
+      0,
+      0,
+      0,
+      initialAppsNonce
+    );
+    return {
+      [AssetType.ETH]: ethFreeBalance
+    };
+  }
+}
+
+/**
+ * Used in standardizing how to set/get app instances within a channel according
+ * to their correct status.
+ */
+export enum APP_INSTANCE_STATUS {
+  INSTALLED = "installed",
+  PROPOSED = "proposed"
+}
+
+/**
+ * Note: this class itelf does not hold any meaningful state either.
+ * It encapsulates the operations performed on relevant appInstances and
+ * abstracts the persistence to the store service.
+ */
+export class Channels {
+  /**
+   * A convenience lookup table from a set of owners to multisig address.
+   */
+  private readonly ownersToMultisigAddress = {};
+
+  /**
+   * A convenience lookup table from appInstance ID to multisig address.
+   */
+  private readonly appInstanceIdToMultisigAddress = {};
+
+  /**
+   * @param selfAddress The address of the account being used with the Node.
+   * @param store
+   * @param multisigKeyPrefix The prefix to add to the key being used
+   *        for indexing multisig addresses according to the execution
+   *        environment.
+   */
+  constructor(
+    public readonly selfAddress: Address,
+    private readonly store: IStoreService,
+    private readonly multisigKeyPrefix: string
+  ) {}
+
+  /**
+   * Called when a new multisig is created for a set of owners.
```

Comment:
AIUI this is called *to* create a new multisig (i.e. not in response to someone else creating a multisig)

Reply:
Correct.

Does the description throw you off from that interpretation? If so, what would be a better description?

Reply:
Actually yeah, the wording can be better. Will update to:
```
Called to create a new multisig for a set of owners.
```